sklearn/cross_decomposition/pls_.py

```diff
@@ -287,6 +290,9 @@ def fit(self, X, Y):
                 self.n_iter_.append(n_iter_)
             elif self.algorithm == "svd":
                 x_weights, y_weights = _svd_cross_product(X=Xk, Y=Yk)
+            # Forces sign stability of x_weights and y_weights
+            x_weights, y_weights = svd_flip(x_weights, y_weights.T)
+            y_weights = y_weights.T
```

Comment:
It is indeed needed as both algorithm have platform dependent sign indeterminacy